02-peer-protocol.md

```diff
@@ -1508,23 +1509,68 @@ Closing happens in two stages:
+        |       |----------------------------->|       |
+        |       |           shutdown(scriptB1) |       |
+        |       |<-----------------------------|       |
         |       |                              |       |
         |       | <complete all pending HTLCs> |       |
-        |   A   |                 ...          |   B   |
+        |   A   |             ....             |   B   |
         |       |                              |       |
-        |       |--(3)-- closing_signed  F1--->|       |
-        |       |<-(4)-- closing_signed  F2----|       |
-        |       |              ...             |       |
-        |       |--(?)-- closing_signed  Fn--->|       |
-        |       |<-(?)-- closing_signed  Fn----|       |
+        |       | closing_complete             |       |
+        |       |----------------------------->|       |
+        |       |             closing_complete |       |
+        |       |<-----------------------------|       |
+        |       |                  closing_sig |       |
+        |       |<-----------------------------|       |
+        |       | closing_sig                  |       |
+        |       |----------------------------->|       |
+        |       |                              |       |
+        |       |   <A updates their script>   |       |
+        |       |                              |       |
+        |       | shutdown(scriptA2)           |       |
+        |       |----------------------------->|       |
+        |       |           shutdown(scriptB1) |       | (*) Bob doesn't update his script
+        |       |<-----------------------------|       |
+        |       | closing_complete             |       |
+        |       |----------------------------->|       |
+        |       |                  closing_sig |       |
+        |       |<-----------------------------|       |
+        |       |                              |       |
+        |       |   <Bob updates his script>   |       |
+        |       |                              |       |
+        |       |           shutdown(scriptB2) |       |
+        |       |<-----------------------------|       |
+        |       | shutdown(scriptA2)           |       |
+        |       |----------------------------->|       |
+        |       |             closing_complete |       |
+        |       |          <-------------------|       |
+        |       |                              |       |
+        |       |  <Alice updates her script>  |       | (*) This is a concurrent update while Bob is sending closing_complete
+        |       |                              |       |
+        |       | shutdown(scriptA3)           |       |
+        |       |------------------->          |       |
+        |       |   closing_complete           |       |
+        |       |<-------------------          |       | (*) A doesn't answer with closing_sig because B's sig doesn't use scriptA3
```

Comment:
> but it's more complex because that means Alice must remember the state she had when she sent shutdown(scriptA2), whereas it's simpler to drop the old state as soon as you send shutdown.

Yeah this does look cleaner.

> That seems dangerous, because it introduces a risk of deadlock: if Bob never sends closing_complete, Alice would never be allowed to send a new shutdown?

Could you clarity a bit? What prevents Alice from sending a new `shutdown` given that node can now send multiple `shutdown` msgs?